oauthenticator/generic.py

```diff
@@ -100,7 +100,7 @@ class GenericOAuthenticator(OAuthenticator):
     async def authenticate(self, handler, data=None):
         code = handler.get_argument("code")
         # TODO: Configure the curl_httpclient for tornado
-        http_client = AsyncHTTPClient()
+        http_client = AsyncHTTPClient(force_instance=True, defaults=dict(validate_cert=self.tls_verify))
```

Comment:
can you remove `force_instance=True` here? after that, I'm happy to merge this now that I understand.

A comment regarding the fact that we specifically need to override the defaults if internal_ssl is enabled could also help keep track of this in the future.

Reply:
Let me verify, but I believe the force_instance is required.